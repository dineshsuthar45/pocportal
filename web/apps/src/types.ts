import { InputProps, TypographyProps } from "@mui/material";
import { CSSProperties, ChangeEvent } from "react";

export interface IThemeSchema {
  palette: any;
  typography: any;
  components?: any;
}

export interface IGlobalTableProps {
  row: any;
  list: any;
}

export interface IGlobalButtonProps {
  label: any;
  sublabel?: string;
  disabled?: boolean;
  onClick?: () => void;
  style?: CSSProperties;
  type?: string;
}

export interface IGlobalImgProps {
  img: string;
  alt: string;
  style?: CSSProperties;
}

export interface IGlobaltextFieldProps {
  id?: string;
  value: string;
  label: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
  name: string;
  errmsg?: string;
  type?: string;
  disabled?: boolean;
  style?: CSSProperties;
  placeholder?: string;
  showPassword?: boolean;
  InputProps?: InputProps;
}

export interface IGlobalTypoProps {
  id?: number;
  variant: TypographyProps["variant"];
  label: string;
  sublabel?: string;
  style?: CSSProperties;
}
