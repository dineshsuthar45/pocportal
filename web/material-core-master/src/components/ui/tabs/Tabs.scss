.tabs-root {
	&.vertical {
		display: flex;
		flex-grow: 1;

		.MuiTabs-vertical {
			border-right: 1px solid transparent;
			border-color: var(--m-colour-greyScale-200);
		}

		&.flip {
			flex-direction: row-reverse;

			.MuiTabs-vertical {
				border-right: none;
				border-left: 1px solid transparent;
				border-color: var(--m-colour-greyScale-200);
			}

			.MuiTabs-indicator {
				left: 0px;
			}
		}
	}

	&.dark {
		.MuiTabs-vertical {
			border-color: var(--m-colour-greyScale-600);
		}

		&.flip {
			.MuiTabs-vertical {
				border-right: none;
				border-left: 1px solid transparent;
				border-color: var(--m-colour-greyScale-600);
			}
		}
	}

	.medium {
		&,
		.MuiButtonBase-root.MuiTab-root {
			min-height: 36px;
			padding: 0px;
		}

		button {
			font-size: 13px;
		}

		.MuiTabs-indicator {
			padding-bottom: 3px;
		}
	}

	.small {
		&,
		.MuiButtonBase-root.MuiTab-root {
			min-height: 30px;
			padding: 0px;
		}

		button {
			font-size: 11px;
		}
	}
}
