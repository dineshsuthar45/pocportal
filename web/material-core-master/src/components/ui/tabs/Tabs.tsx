import React from 'react';
import { Box, Tabs as MuiTabs, useTheme } from '@mui/material';
import classNames from 'classnames';
import SwipeableViews from 'react-swipeable-views';
import { MPanel } from '../internal/panel';
import { MTab } from '../internal/tab';
import './Tabs.scss';
import { TabsProps } from './Tabs.types';

export const MTabs = (props: TabsProps) => {
	const theme = useTheme();
	const { options, value, onChange, component: Component, ...otherProps } = props;

	const tabOptions = options.map((option, index) => {
		return <MTab key={index} {...option} color={props.color} disabled={props.disabled}></MTab>;
	});

	const tabPanels = options.map((option, index) => {
		return (
			<MPanel value={value} index={index} key={index}>
				{option.view}
			</MPanel>
		);
	});

	return (
		<Box
			className={classNames('tabs-root', theme.palette.mode, {
				vertical: props.orientation === 'vertical',
				flip: props.orientation === 'vertical' && props.verticalflip,
			})}
			sx={{ height: props.maxHeight }}
		>
			<MuiTabs
				className={classNames(props.size)}
				disabled={props.disabled}
				variant={props.variant}
				centered={props.centered}
				indicatorColor={props.color}
				orientation={props.orientation}
				textColor={props.color}
				value={value}
				component={Component}
				{...otherProps}
				onChange={(event, value) => onChange(Number(value))}
			>
				{tabOptions}
			</MuiTabs>
			<SwipeableViews index={value}>{tabPanels}</SwipeableViews>
		</Box>
	);
};

MTabs.defaultProps = {
	centered: false,
	color: 'primary',
	disabled: false,
	orientation: 'horizontal',
	scrollButtons: 'auto',
	selectionFollowsFocus: false,
	size: 'medium',
	variant: 'standard',
	visibleScrollbar: false,
	verticalflip: false,
};
