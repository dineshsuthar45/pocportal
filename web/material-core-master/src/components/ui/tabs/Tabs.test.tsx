import { render } from '@testing-library/react';
import React from 'react';
import { MTabs } from './Tabs';
import { TabsProps } from './Tabs.types';

describe('Tabs', () => {
	const renderComponent = () => render(<MTabs {...props} />);
	let props: TabsProps;

	beforeEach(() => {
		props = {
			options: [],
			onChange: () => {},
			value: 0,
		};
	});

	it('should render correctly', () => {
		const { container } = renderComponent();
		expect(container).toMatchSnapshot();
	});

	it('should render correctly with options', () => {
		props.options = [
			{
				value: 0,
				label: 'Tab label 1',
			},
			{
				value: 1,
				label: 'Tab label 2',
			},
		];
		const { container } = renderComponent();
		expect(container).toMatchSnapshot();
	});
});
