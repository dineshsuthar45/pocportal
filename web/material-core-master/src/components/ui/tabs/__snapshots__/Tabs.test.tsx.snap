// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render correctly 1`] = `
<div>
  <div
    class="tabs-root light MuiBox-root css-0"
  >
    <div
      class="MuiTabs-root medium css-1ujnqem-MuiTabs-root"
      color="primary"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        />
        <span
          class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      style="overflow-x: hidden;"
    >
      <div
        class="react-swipeable-view-container"
        style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
      />
    </div>
  </div>
</div>
`;

exports[`Tabs should render correctly with options 1`] = `
<div>
  <div
    class="tabs-root light MuiBox-root css-0"
  >
    <div
      class="MuiTabs-root medium css-1ujnqem-MuiTabs-root"
      color="primary"
    >
      <div
        class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        style="overflow: hidden; margin-bottom: 0px;"
      >
        <div
          class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          role="tablist"
        >
          <button
            aria-selected="true"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected tab-root light css-bqvrnz-MuiButtonBase-root-MuiTab-root"
            color="primary"
            role="tab"
            tabindex="0"
            type="button"
          >
            Tab label 1
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-selected="false"
            class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary tab-root light css-bqvrnz-MuiButtonBase-root-MuiTab-root"
            color="primary"
            role="tab"
            tabindex="-1"
            type="button"
          >
            Tab label 2
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <span
          class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
          style="left: 0px; width: 0px;"
        />
      </div>
    </div>
    <div
      style="overflow-x: hidden;"
    >
      <div
        class="react-swipeable-view-container"
        style="flex-direction: row; transition: all 0s ease 0s; direction: ltr; display: flex; will-change: transform;"
      >
        <div
          aria-hidden="false"
          data-swipeable="true"
          style="width: 100%; flex-shrink: 0; overflow: auto;"
        >
          <div
            class="panel-root"
            id="0"
          >
            <div
              class="MuiBox-root css-0"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
